### PR TITLE
[Merged by Bors] - Add peer score adjustment msgs

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

N/A

## Proposed Changes

This PR adds the `msg` field to `Peer score adjusted` log messages. These `msg` fields help identify *why* a peer was banned.

Example:

```
Jan 11 04:18:48.096 DEBG Peer score adjusted                     score: -100.00, peer_id: 16Uiu2HAmQskxKWWGYfginwZ51n5uDbhvjHYnvASK7PZ5gBdLmzWj, msg: attn_unknown_head, service: libp2p
Jan 11 04:18:48.096 DEBG Peer score adjusted                     score: -27.86, peer_id: 16Uiu2HAmA7cCb3MemVDbK3MHZoSb7VN3cFUG3vuSZgnGesuVhPDE, msg: sync_past_slot, service: libp2p
Jan 11 04:18:48.096 DEBG Peer score adjusted                     score: -100.00, peer_id: 16Uiu2HAmQskxKWWGYfginwZ51n5uDbhvjHYnvASK7PZ5gBdLmzWj, msg: attn_unknown_head, service: libp2p
Jan 11 04:18:48.096 DEBG Peer score adjusted                     score: -28.86, peer_id: 16Uiu2HAmA7cCb3MemVDbK3MHZoSb7VN3cFUG3vuSZgnGesuVhPDE, msg: sync_past_slot, service: libp2p
Jan 11 04:18:48.096 DEBG Peer score adjusted                     score: -29.86, peer_id: 16Uiu2HAmA7cCb3MemVDbK3MHZoSb7VN3cFUG3vuSZgnGesuVhPDE, msg: sync_past_slot, service: libp2p
```

There is also a `libp2p_report_peer_msgs_total` metrics which allows us to see count of reports per `msg` tag. 

## Additional Info

NA

### CODE REVIEW

File: beacon_node/network/src/beacon_processor/worker/gossip_methods.rs [hunk 1]
```diff
@@ -734,16 +735,23 @@ impl<T: BeaconChainTypes> Worker<T> {
                 self.send_sync_message(SyncMessage::UnknownBlock(peer_id, block));
                 return None;
             }
+            Err(e @ BlockError::BeaconChainError(_)) => {
+                debug!(
+                    self.log,
+                    "Gossip block beacon chain error";
+                    "error" => ?e,
+                );
```
Comment (michaelsproul) [inline beacon_node/network/src/beacon_processor/worker/gossip_methods.rs:747]: For consistency I think we should we also propagate an `Ignore` here? What does the network stack do if we never get back to it about a message?

**Comment by michaelsproul** [comment]
This will clash with #2894, so we'll have to merge them in separate batches (or roll them into one PR)

File: beacon_node/network/src/beacon_processor/worker/gossip_methods.rs (hunk 1 above)
Comment (paulhauner) [reply beacon_node/network/src/beacon_processor/worker/gossip_methods.rs:747]: I think nothing is the same as in `Ignore`, but I'll add it here for consistency. Thanks!

**Review by michaelsproul — APPROVED** [review]
LGTM!

Happy to merge pending a successful Clippy run post enum removal

**Pull request closed without merging**